File: matlab_examples/FrequencyUpdated.m
function FrequencyUpdated(src, event)
    frequency = double(event.CurrentFrequency);  
    step = int32(event.CurrentStep);
    lineLength = fprintf('Frequency (Hz): %.2f - Step %d/%d', frequency, step, int32(event.TotalSteps));
    pause(0.001);
    if (step < int32(event.TotalSteps))
        fprintf(repmat('\b', 1 ,lineLength));
    else
        fprintf('\n');
    end     
end

File: python_examples/FrequencyUpdated.py
def frequencyupdated(source, args):
    if args.CurrentStep < args.TotalSteps:
        print('\rFrequency (Hz): {0:.2f} - Step {1} / {2}'.format(args.CurrentFrequency, args.CurrentStep,
                                                                  args.TotalSteps), end='')
    else:
        print('\rFrequency (Hz): {0:.2f} - Step {1} / {2}'.format(args.CurrentFrequency, args.TotalSteps,
                                                                  args.TotalSteps), end='')
        print('')